libs/server/ArgSlice/ArgSliceUtils.cs

```diff
@@ -15,5 +17,139 @@ public static class ArgSliceUtils
+            if (slice.Span[0] == type)
+            {
+                qStart = true;
+                start = 1;
+            }
+
+            for (var j = start; j < slice.Span.Length - shift - 1; ++j)
+            {
+                if (!qStart)
+                {
+                    if (slice.Span[j + shift] == type)
+                    {
+                        qStart = true;
+                        shift++;
+                        slice.Span[j] = slice.Span[j + 1];
+                    }
+                    continue;
+                }
+
+                if (slice.Span[j + shift] != '\\')
+                {
+                    // If we shifted earlier, we need to shift the rest too.
+                    if (shift > 0)
+                        slice.Span[j] = slice.Span[j + shift];
+                    continue;
+                }
+
+                // ' context recognize only this particular sequence
+                if (type == '\'')
+                {
+                    if (slice.Span[j + shift + 1] == '\'')
+                        shift++;
+                    if (shift > 0)
+                        slice.Span[j] = slice.Span[j + shift];
+                    continue;
+                }
+
+                // Process escapes
+                shift++;
+                var c = acceptUppercaseEscapes ?
+                            (char)AsciiUtils.ToLower(slice.Span[j + shift]) :
+                            (char)slice.Span[j + shift];
+
+                switch (c)
+                {
+                    case 'a':
+                        c = '\a';
+                        break;
+                    case 'b':
+                        c = '\b';
+                        break;
+                    case 'n':
+                        c = '\n';
+                        break;
+                    case 'r':
+                        c = '\r';
+                        break;
+                    case 't':
+                        c = '\t';
+                        break;
+                    case 'x':
+                        if (j + shift + 2 < slice.Span.Length)
+                        {
+                            try
+                            {
+                                var res = Convert.FromHexString(Encoding.ASCII.GetString(slice.Span.Slice(j + shift + 1, 2)));
+                                c = (char)res[0];
+                                shift += 2;
+                                break;
+                            }
+                            catch
+                            {
+
+                            }
+                        }
+                        break;
+                    default:
+                        break;
+                }
+
+                slice.Span[j] = (byte)c;
+            }
+
+            // Zero unnecessary chars
+            for (var l = slice.Span.Length - shift - start; l < slice.Span.Length - 1; ++l)
```

Comment:
`slice.Span[(slice.Span.Length - shift - start)..(slice.Span.Length - 1)].Clear()` is equivalent.